osu.Game/Database/BackgroundDataStoreProcessor.cs

```diff
@@ -376,6 +376,66 @@ private void convertLegacyTotalScoreToStandardised()
             completeNotification(notification, processedCount, scoreIds.Count, failedCount);
         }
 
+        private void upgradeScoreRanks()
+        {
+            Logger.Log("Querying for scores that need rank upgrades...");
+
+            HashSet<Guid> scoreIds = realmAccess.Run(r => new HashSet<Guid>(
+                r.All<ScoreInfo>()
+                 .Where(s => s.TotalScoreVersion < LegacyScoreEncoder.LATEST_VERSION)
+                 .AsEnumerable()
+                 // must be done after materialisation, as realm doesn't support
+                 // filtering on nested property predicates or projection via `.Select()`
+                 .Where(s => s.Ruleset.IsLegacyRuleset())
```

Comment:
Right, I removed this thinking it was checking legacy scores, but turns out this is important I guess.